### PR TITLE
Cancel multi method charge

### PR DESCRIPTION
**Pull request by wallaceSF**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| **Issue**         | https://mundipagg.atlassian.net/browse/MOD-930
| **What?**         | Create structure to cancel charge paid (multimethod when exist paid between failed.)
| **Why?**          | Because an charge stay pay in mundipagg

### CODE REVIEW

**Comment by wallaceSF** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Coverage decreased per file
===========================
- src/Kernel/Aggregates/Charge.php  -1
         

Clones added
============
- src/Webhook/Services/ChargeOrderService.php  1
- src/Kernel/Services/ChargeService.php  2
- src/Webhook/Services/ChargeRecurrenceService.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mundipagg/ecommerce-module-core/pullRequest?prid=4938457&bid=16329075)

File: src/Webhook/Services/ChargeHandlerService.php [hunk 1]
```diff
@@ -45,6 +50,47 @@ public function build($component)
     public function handle(Webhook $webhook)
     {
         $this->build($webhook->getComponent());
-        return $this->listChargeHandleService->handle($webhook);
+        $multiMeiosCanceled = $this->tryCancelMultiMethods($webhook);
```
Comment (michelpl) [inline src/Webhook/Services/ChargeHandlerService.php]: Traduzir variável

File: src/Webhook/Services/ChargeHandlerService.php [hunk 2]
```diff
@@ -45,6 +50,47 @@ public function build($component)
...
+
+        return array_merge(
+            $this->listChargeHandleService->handle($webhook),
+            $multiMeiosCanceled
```
Comment (michelpl) [inline src/Webhook/Services/ChargeHandlerService.php]: Traduzir variável

File: src/Webhook/Services/ChargeOrderService.php [hunk 3]
```diff
@@ -82,8 +84,11 @@ protected function handlePaid(Webhook $webhook)
         $platformOrder->save();
 
         $returnMessage = $this->prepareReturnMessage($charge);
+
+        $response = $this->tryCancelMultiMethodsWithOrder();
+
         $result = [
-            "message" => $returnMessage,
+            "message" => $returnMessage . '  '. $response,
```
Comment (michelpl) [inline src/Webhook/Services/ChargeOrderService.php]: Adicionar espaço entre aspas e ponto

File: src/Webhook/Services/ChargeHandlerService.php (hunk 1 above)
Comment (GabrielDeveloper) [inline src/Webhook/Services/ChargeHandlerService.php]: Trocar multimeios para multimethods

File: src/Webhook/Services/ChargeHandlerService.php (hunk 2 above)
Comment (GabrielDeveloper) [inline src/Webhook/Services/ChargeHandlerService.php]: Trocar multimeios para multimethods

**Review by michelpl — CHANGES_REQUESTED** [review]

**Review by GabrielDeveloper — APPROVED** [review]

**Comment by wallaceSF** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Coverage decreased per file
===========================
- src/Kernel/Aggregates/Charge.php  -1
         

Clones added
============
- src/Webhook/Services/ChargeRecurrenceService.php  1
- src/Kernel/Services/ChargeService.php  2
- src/Webhook/Services/ChargeOrderService.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mundipagg/ecommerce-module-core/pullRequest?prid=4938457&bid=16329075)

**Comment by wallaceSF** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Clones added
============
- src/Kernel/Services/ChargeService.php  2
- src/Webhook/Services/ChargeOrderService.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mundipagg/ecommerce-module-core/pullRequest?prid=4938457&bid=16329075)

**Pull request merged**